lib/Model/ConvertFunctionType.cpp

```diff
@@ -0,0 +1,633 @@
+
+  model::RawFunctionType Result;
+  for (size_t ArgumentIdx = 0; ArgumentIdx < Arguments.size(); ++ArgumentIdx) {
+    auto &ArgumentStorage = Arguments[ArgumentIdx];
+    const auto &ArgumentType = Function.Arguments.at(ArgumentIdx).Type;
+    if (!ArgumentStorage.Registers.empty()) {
+      // Handle the registers
+      auto OriginalName = Function.Arguments.at(ArgumentIdx).CustomName;
+      for (size_t Index = 0; auto Register : ArgumentStorage.Registers) {
+        auto FinalName = OriginalName;
+        if (ArgumentStorage.Registers.size() > 1 && !FinalName.empty())
+          FinalName += "_part_" + std::to_string(++Index) + "_out_of_"
+                       + std::to_string(ArgumentStorage.Registers.size());
+        model::NamedTypedRegister Argument(Register);
+        Argument.Type = chooseArgumentType(ArgumentType,
+                                           Register,
+                                           ArgumentStorage.Registers,
+                                           TheBinary);
+        Argument.CustomName = FinalName;
+        Result.Arguments.insert(Argument);
+      }
+    }
+
+    if (ArgumentStorage.SizeOnStack != 0) {
+      // Handle the stack
+      auto ArgumentIterator = Function.Arguments.find(ArgumentIdx);
+      revng_assert(ArgumentIterator != Function.Arguments.end());
+      const model::Argument &Argument = *ArgumentIterator;
+
+      /// TODO: handle stack arguments properly.
+      /// \note: different ABIs could use different stack types.
+      /// \sa: `clrcall` ABI.
+    }
+  }
+
+  if (!Function.ReturnType.isVoid()) {
+    using CC = CallingConvention;
+    auto ReturnValue = distributeReturnValue<CC>(Function.ReturnType);
+    if (ReturnValue == std::nullopt)
+      return std::nullopt;
+
+    if (!ReturnValue->Registers.empty()) {
+      // Handle a register-based return value.
+      for (model::Register::Values Register : ReturnValue->Registers) {
+        model::TypedRegister ReturnValueRegister;
+        ReturnValueRegister.Location = Register;
+        ReturnValueRegister.Type = chooseArgumentType(Function.ReturnType,
+                                                      Register,
+                                                      ReturnValue->Registers,
+                                                      TheBinary);
+
+        Result.ReturnValues.insert(std::move(ReturnValueRegister));
+      }
+
+      // Try and recover types from the struct if possible
+      if (Function.ReturnType.Qualifiers.empty()) {
+        const model::Type *Type = Function.ReturnType.UnqualifiedType.get();
+        const auto *Struct = llvm::dyn_cast<model::StructType>(Type);
+        if (Struct && Struct->Fields.size() == Result.ReturnValues.size()) {
+          using RegisterEnum = model::Register::Values;
+          SmallMap<RegisterEnum, model::QualifiedType, 4> RecoveredTypes;
+          size_t StructOffset = 0;
+          for (size_t Index = 0; Index < Struct->Fields.size(); ++Index) {
```

Comment:
When you have a large loop with many `break`/`continue` it's almost always a good idea to outline it in a function and use `return`.

Reply:
Done (https://github.com/revng/revng/pull/98/commits/e228a3d1eaa527022056bfee871081ece5f52d4e).